src/zenml/orchestrators/step_runner.py

```diff
@@ -447,14 +447,22 @@ def _load_input_artifact(
             # we use the datatype of the stored artifact
             data_type = source_utils.load(artifact.data_type)
 
```

Comment:
Shall the `_load_artifact_from_uri` also be changed?

Reply:
As far as I can see, there is no need. `_load_artifact_from_uri` accepts an artifact store instance as an input and passes it to the constructor of the materializer. 

Any materializer that uses `self._artifact_store` will have access to it and any materializer that uses `fileio` will have access to the correct filesystem because the artifact store is already initialized at this point.